os/kernel/binary_manager/binary_manager_recovery.c

```diff
@@ -334,14 +337,16 @@ void binary_manager_recovery(int bin_idx)
 	/* If a fault happens in common or user binaries, we need to reload the library and all user binaries */
 	int bidx;
 	int bin_count = binary_manager_get_ucount();
+
+	BIN_STATE(BM_CMNLIB_IDX) = BINARY_FAULT;
```

Comment:
Please leave a comment why you change the state of common lib unlike others.

Reply:
As we discussed on the chat, let's move this into deactivate function like other and do changing the state for common there. Then, it's much better to understand and it's better to maintain.